### ISSUE TITLE
SEO friendliness

### ISSUE THREAD

**Issue by srid** [tfausak/haskell-knowledge-base#2]
Looks like Obsidian's published sites don't prerender. I would think that this means that, depending on how successful search engines' JS evaluation is, the haskell KB may not have a great SEO score. 

```
❯ curl https://publish.obsidian.md/haskell/README
```

This produces a body with only script, and no content:

```html
<body class="theme-dark">
    <script>window.siteInfo = { "uid": "c3ea0e362c08cbd3e6de985ca5a3156e", "host": "publish-01.obsidian.md", "status": "active", "slug": "haskell", "redirect": null, "customurl": null };</script>
    <script src="/lib/purify.min.js"></script>
</body>
```

**Comment by srid** [comment]
> depending on how successful search engines' JS evaluation is

Google attempts to do this, but not other search engines like Bing or DuckDuckGo. So the published site is not even reachable from them.

Eg: https://publish.obsidian.md/alexisrondeau/4+New+Things+I+Have+Noticed+About+Berlin

![image](https://user-images.githubusercontent.com/3998/115279890-40e3fb00-a115-11eb-88fd-05b2e326a368.png)

![image](https://user-images.githubusercontent.com/3998/115279963-5bb66f80-a115-11eb-928f-6034edc27063.png)

**Comment by tfausak** [comment]
I'm not sure what we can do about this. We could publish with something other than Obsidian, but that would take some research. We could also serve different sites to users and crawlers, but that would take a fair amount of work. (Plus we'd still have to figure out how to pre-render the content.) 

It's clearly not ideal, but the sources in GitHub should get picked up by search engines.

**Comment by tfausak** [comment]
Another option is to position this repository as a data source that others are encouraged to publish, perhaps through other means like Neuron 😉 Still, it would be nice for the "official" approach to have good SEO (and be accessible, and work on mobile, and ...).

**Comment by srid** [comment]
That's already the case I assume? That this repository will remain future-proof, by staying as nothing more than a directory of Markdown notes linked to one another using the increasingly-becoming-standard `[[..]]` wiki-links?

I do like to create a demo static site[^fn] (@TheMatten expressed some interest as well) that uses the files on this repo. Not necessarily based on neuron (since that's limited to Zettelkasten'ish UI), but doing a completely customizable HTML/CSS (using [ema](https://github.com/srid/ema)), and with a left sidebar like that of Obsidian site; or the [uptree style](https://alien-psychology.zettel.page/S02E23) of neuron, if that's more desirable) - with potential for doing cool mind-map based visualizations. It will all be 100% Haskell.

[^fn]: which would be great to use as official KB. But if not, it will atleast remain as an example for the upcoming `ema` static site generator.

**Comment by tfausak** [comment]
> That's already the case I assume?

Yes, that's correct. What I meant to say was that I could call that out explicitly. Something that explains this is where the data lives, and there's a default visualization over here, but you may also want to check out these other visualizations.

**Comment by srid** [comment]
That makes sense. 

We might as well rename this repo to, say, `haskell-kb` (which matches a site hostname like `kb.haskell.org` where KB = knowledge base) so that it doesn't give an impression of being specific to Obsidian.

As an aside, newcomers to Haskellers may even end up considering it to be a community wiki of some sorts as an aid for learning (hopefully destined to be better in many ways to the current wiki.haskell.org; I already like the [atomicity](https://neuron.zettel.page/atomic) -- which makes linking more significant -- of the content currently populated in this repo), where one can directly edit it on GitHub (which [automatically opens the PR](https://docs.github.com/en/github/managing-files-in-a-repository/editing-files-in-your-repository); and we can add an "edit this page" button on the site, that takes them to GitHub. This trick is used in https://zk.zettel.page/

**Comment by tfausak** [comment]
This was arguably fixed by #6, although that site isn't linked from anywhere yet.

**Comment by tfausak** [comment]
Closing as fixed by #9.
